docs/provider/google-secrets-manager.md

```diff
@@ -1,118 +1,250 @@
-## Google Cloud Secret Manager
```

Comment:
Why did you delete the title? 🤔

Reply:
The title appears twice right now: https://external-secrets.io/latest/provider/google-secrets-manager/

![image](https://github.com/user-attachments/assets/166d9d3d-5a09-4b48-ab67-78f3b6705709)

I deleted this line to keep just the H1 but obviously happy to put it back in if it's there on purpose :-)

Reply:
Ah thank you! 👍